modules/turn/turn.c

```diff
@@ -177,13 +178,17 @@ static bool indication_handler(struct restund_msgctx *ctx, int proto,
 	if (!peer || !data)
 		return true;
 
-	perm = perm_find(al->perms, &peer->v.xor_peer_addr);
+	psa = &peer->v.xor_peer_addr;
+	perm = perm_find(al->perms, psa);
 	if (!perm) {
 		++al->dropc_tx;
 		return true;
 	}
 
-	err = udp_send(al->rel_us, &peer->v.xor_peer_addr, &data->v.data);
+	if (sa_is_loopback(psa) || sa_is_any(psa) || sa_is_linklocal(psa))
```

Comment:
Hmm `sa_is_loopback` checks if the address is equal to `127.0.0.1` but we shouldn't we check whether the address is in the  `127.0.0.0/8` range?

It is good enough in our case as we actually only configure `restund` to listen on `127.0.0.1` specifically; but there are other things on loopback that we do not want to be reachable. E.g. `127.0.0.53` hosts the `systemd-resolved` DNS resolver  on Ubuntu machines and relaying to it could trigger a DoS


restund: https://github.com/creytiv/re/blob/ea0509d6fc9a8beb366dde731885ff7d9b4d2d8b/src/sa/sa.c#L555

coturn: https://github.com/coturn/coturn/blob/060bf187879fd1a6386012f4c5a7494824ebe5c8/src/client/ns_turn_ioaddr.c#L478-L497

Reply:
Good points. Added.